### PR TITLE
refactor: Move messages and events code from dispatcher pkg

### LINKED ISSUE #342: Refactor dispatcher pkg
Currently dispatcher pkg has functions related to dispatcher, messages and events (added to avoid cyclical dependency). These need to be refactored.

### PR DESCRIPTION
**Pull request by rolsonquadras**
- Move messages and events out of dispatcher pkg to separate pkgs

Closes #342 

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

### CODE REVIEW

File: pkg/client/didexchange/client.go [hunk 1]
```diff
@@ -15,9 +15,11 @@ import (
 
 	"github.com/google/uuid"
 
-	"github.com/hyperledger/aries-framework-go/pkg/didcomm/dispatcher"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/model"
```
Comment (troyronda) [inline pkg/client/didexchange/client.go]: What's the rationale for a `model` package?

Comment (rolsonquadras) [reply pkg/client/didexchange/client.go]: It contains DIDCommMsg and Destination. These are used handling the messages throughout the framework. ie, pass to handle from input, pass in event messages.

May be messages would make more sense ? Any suggestions ?

File: pkg/client/didexchange/client.go [hunk 2]
```diff
@@ -15,9 +15,11 @@ import (
...
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/event"
```
Comment (troyronda) [inline pkg/client/didexchange/client.go]: I see a common folder above. Should this be didcomm/common/event & didcomm/common/service ?

Comment (rolsonquadras) [reply pkg/client/didexchange/client.go]: As discussed, moved messages and event relates code to "didcomm/common/service".

Also, to remove cyclical dependency, created a new [Handler function](https://github.com/hyperledger/aries-framework-go/pull/444/files#diff-75342fbf986feea5a6bb6a33d0caff76R11) and used it in [dispatcher.Service](https://github.com/hyperledger/aries-framework-go/pull/444/files#diff-6a4454074ff31ccb82bdbcc343255467R17).

File: pkg/didcomm/common/service/service.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+package service
+
+// Handler provides protocol service handle api.
+type Handler interface {
+	Handle(msg DIDCommMsg) error
+}
+
+// DIDCommService defines service APIs.
+type DIDCommService interface {
```
Comment (llorllale) [inline pkg/didcomm/common/service/service.go]: Consider removing the "Service" suffix since this is now in the `service` package.

**Comment by troyronda** [comment]
@rolsonquadras this PR will need rebase.

**Comment by rolsonquadras** [comment]
> Names are now much better aligned with the semantics

[Updated](https://github.com/hyperledger/aries-framework-go/compare/3718252dfb4c1793dba23c241e3c1dd59c75b6c7..8172ea300689af40acb68c653147bafac9847bd3#diff-75342fbf986feea5a6bb6a33d0caff76R15).

**Comment by rolsonquadras** [comment]
> @rolsonquadras this PR will need rebase.

Rebased. Now no merge conflicts.

**Pull request merged**